### ISSUE TITLE
jack2 is incompatible with Java Audio on OS X [again]

### ISSUE THREAD

**Issue by eclab** [jackaudio/jack2#419]
This is a reopening of #378 which was closed because a developer couldn't be bothered to look up what Java Audio was.  So let me try again.  After installing jack2 on OS X, any code which uses the javax.sound.sampled package to dump data vis a a SourceDataLine, and its underlying sound output driver, will fail with a java error.  

I can't provide the error message because, after spending several hours to remove jack2 (its uninstaller is very broken), I'm not prepared to install it again just to semipermanently break my development environment again for several hours.  :-(  But as a javax.sound.sampled developer, I *can* verify that jack2 appears to completely bork Java audio.

**Comment by 7890** [comment]
There is nothing we can do for fixing jack2 on OSX. The package is static.
Once we have a new binary release package for OSX we can look at issues related to that release. To track down issues with 3rd party software, we will need precise information about the failing process. Please check back again later for a new OSX release.

**Comment by 7890** [comment]
@eclab If JACK isn't seen by Java it could be because JackRouter doesn't work on recent OSX.

https://github.com/jackaudio/jack2/issues/305
> @jefftime In my understanding, you need jackrouter to communicate between CoreAudio application and jack application, but traditional jackrouter doesn't work on recent version of MacOS (maybe after El Capitan)

IIUC javax.sound will offer whatever CoreAudio is presenting.
Using a small program "MediaTypes.java" from here
https://stackoverflow.com/questions/5304001/javasound-mixer-with-both-ports-and-datalines
you can check which sinks Java can see.

As far as I understand making JackRouter to work with recent OSX isn't an easy fix (the OS changed something that affects how JackRouter can work) so don't expect a solution anytime soon.

That said it's obvious that JACK without the router is massively less powerful. The issue with JackRouter is known
.
tl;dr This issue is most probably not related to Java at all.

**Comment by 7890** [comment]
Also see https://github.com/jackaudio/jack2/issues/84
